### DISCUSSION TITLE
Trigger email on CustomOrderState Transition

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1674] [Q&A]
When adding email handler to OrderStateTrasition for a custom state it throwing a typescript error.

"**This condition will always return 'false' since the types 'OrderState' and '"CustomerKYCVerified"' have no overlap.**"

    export const kycDocumentsUploadPendingHandler = new EmailEventListener('kyc-received')
    .on(OrderStateTransitionEvent)
    .filter(
        event =>
            event.toState === 'CustomerKYCVerified' && event.fromState === 'PaymentSettled' && !!event.order.customer,
    )

**Comment by michaelbromley** [comment]
Hi,
This is happening just because TypeScript has no knowledge of your custom OrderState `CustomerKYCVerified`. Right now in TypeScript there's no way to extend a built-in union type that I know of. So the correct solution here is to make a type assertion about the first `event.toState`,

```ts
type MyOrderState = OrderState | 'CustomerKYCVerified';

// ...
event => (event.toState as MyOrderState) = 'CustomerKYCVerified' ...
```
